pkg/trace/api/api.go

```diff
@@ -132,10 +133,26 @@ func (r *HTTPReceiver) Start() {
 		go func() {
 			defer watchdog.LogOnPanic()
 			r.server.Serve(ln)
+			ln.Close()
 		}()
 		log.Infof("Listening for traces at unix://%s", path)
 	}
 
+	if path := mainconfig.Datadog.GetString("apm_config.windows_pipe_name"); path != "" {
+		pipepath := `\\.\pipe\` + path
+		bufferSize := mainconfig.Datadog.GetInt("apm_config.windows_pipe_buffer_size")
+		ln, err := listenPipe(pipepath, bufferSize)
+		if err != nil {
+			killProcess("Error creating %q named pipe: %v", pipepath, err)
+		}
+		go func() {
+			defer watchdog.LogOnPanic()
+			r.server.Serve(ln)
```

Comment:
I do not know how the client of this API sends the data, but for DogStatsd, I had to handle the case where a message is partially written in the read buffer at https://github.com/DataDog/datadog-agent/blob/7.22.1/pkg/dogstatsd/listeners/named_pipe_windows.go#L143.

Here is an example:
 - The read buffer size is 150 bytes
 - The client of this API send a first message of 100 bytes
 - The client of this API send a second message of 100 bytes quickly after the first one.
In the case of DogStatsd, `conn.Read(buffer[startWriteIndex:]` returns a buffer of 150 bytes containing the first message and half of the second message.
 
Depending how `r.server.Serve(ln)` is implemented, it may be or not an issue.

Reply:
Thanks for this. We should figure it out during out integration testing. I suspect it should be fine...